src/NexusMods.Paths/Utilities/KnownExtensions.cs

```diff
@@ -1,52 +1,650 @@
+        // Script
+        { Bat, ExtensionCategory.Script },
+        { Cs, ExtensionCategory.Script },
+        { Fx, ExtensionCategory.Script },
+        { Glsl, ExtensionCategory.Script },
+        { Hks, ExtensionCategory.Script },
+        { Hlsl, ExtensionCategory.Script },
+        { Lua, ExtensionCategory.Script },
+        { Py, ExtensionCategory.Script },
+        { Sh, ExtensionCategory.Script },
+        { Ws, ExtensionCategory.Script },
+        // Text
+        { Cfg, ExtensionCategory.Text },
+        { Json, ExtensionCategory.Text },
+        { Ini, ExtensionCategory.Text },
+        { Log, ExtensionCategory.Text },
+        { Md, ExtensionCategory.Text },
+        { Pdf, ExtensionCategory.Text },
+        { Preset, ExtensionCategory.Text },
+        { Txt, ExtensionCategory.Text },
+        { Toml, ExtensionCategory.Text },
+        { Xml, ExtensionCategory.Text },
+        { Yml, ExtensionCategory.Text },
+        { Yaml, ExtensionCategory.Text },
+        { Vmf, ExtensionCategory.Text },
+        // Video
+        { Bik, ExtensionCategory.Video },
+        { Bk2, ExtensionCategory.Video },
+        { Mp4, ExtensionCategory.Video },
+        { Sfd, ExtensionCategory.Video },
+        { Usm, ExtensionCategory.Video },
+        { Wmv, ExtensionCategory.Video },
+    };
+}
+
+/// <summary>
+///     Provides categories for each common extension.
+/// </summary>
+[PublicAPI]
+public enum ExtensionCategory
+{
+    /// <summary>
+    /// Represents archive and compressed file formats.
+    /// </summary>
+    Archive,
+
+    /// <summary>
+    /// Represents archives containing audio data.
+    /// </summary>
+    ArchiveOfAudio,
+
+    /// <summary>
+    /// Represents archives containing texture/image data.
+    /// </summary>
+    ArchiveOfImage,
+
+    /// <summary>
+    /// Represents audio and sound file formats.
+    /// </summary>
+    Audio,
+
+    /// <summary>
+    /// Represents binary.
+    /// </summary>
+    Binary,
+
+    /// <summary>
+    /// Represents database and data storage formats.
+    /// </summary>
+    Database,
+
+    /// <summary>
+    /// Represents executable and application file formats.
+    /// </summary>
+    Executable,
+
+    /// <summary>
+    /// Represents image and graphic file formats.
+    /// </summary>
+    Image,
+
+    /// <summary>
+    /// Represents dynamic library and code library formats.
+    /// </summary>
+    Library,
+
+    /// <summary>
+    /// Represents 3D model and related file formats.
+    /// </summary>
+    Model,
+
+    /// <summary>
+    /// Represents script and source code file formats.
+    /// </summary>
+    Script,
+
+    /// <summary>
+    /// Represents text, configuration, and documentation file formats.
+    /// </summary>
+    Text,
+
+    /// <summary>
+    /// Represents video, movie, and cutscene file formats.
+    /// </summary>
+    Video,
+
+    /// <summary>
+    /// Unknown file format.
+    /// </summary>
+    Unknown,
```

Comment:
- Put this in a separate file.
- Explicitly define the underlying type (use `byte`).
- Explicitly define the underlying values (`0` to _n_).
- Put `Unknown` first with value `0`.
- Remove `ArchiveOfImage` and `ArchiveOfAudio` we don't need to be this specific.
- Remove `Library` and maybe `Binary`.
- Update the `Model` doc-string, 2D models are a thing as well.

The only actual things we need for the UI are

- `Archive`,
- `Audio`,
- `Image`,
- `Video`,
- and `Text`.

`Archive` is also kinda useless because the File Extractor uses signatures (magic bytes) to know if a file is an archive or not, so we won't be using this extension list for archive detection.

Reply:
Alright, addressed some of the feedback.

With regards to the categories. I chose to do it this way for a very specific reason.

The intent here isn't to match the exact categories of the current UI designs in NMA. Instead, the intent is to group the extensions into as many distinct categories as possible. We then map those sub/smaller categories into the bigger categories that Design wishes us to show in the UI. 

That works in a similar way to the 3 layer design system we use for colours in the App. If design wants to make a change to some colours, or icons, it becomes very easy for us to quickly adapt to their changes.